### ISSUE TITLE
Plea for help with accessing nested DOM tree attributes

### ISSUE THREAD

**Issue by I-keep-trying** [cheeriojs/cheerio#1156]
This is most likely not a bug. 

However, I have already tried learning puppeteer, and a couple other web scraping tools, and cheerio is _by far_ the most elegant, well documented, well, it just seems to be 99.9% working immediately with minimal fuss. 

With all that said, I just cannot get past an issue that persists no matter what tool i try, and no amount of googling and nobody on stack overflow seems to have the same problem. Maybe I'm just asking the wrong question, I will try to explain myself the best I can, and attach a picture. 

All the tutorials, without exception, and I have done a lot of them, just ignore the fact that A) not all websites follow a nice tree heirarchy that puts the data you want in a nice stair-stepped layout with all related child nodes underneath a parent attribute. Sometimes the related data is a sibling, and there is nothing you can do about that, how can that be handled programmatically? And B) Sometimes the related child nodes are children of children, or even further down the tree, of a parent element, with no identifying attributes of their own. The DOM tree is only concerned with making the site look a certain way, unfortunately for us. Therefore, I think it's actually not uncommon for data to be found in elements that are more than one level deep, like grandchildren or great-grandchildren elements, if that is a term??? 

![metadata](https://user-images.githubusercontent.com/26046173/36872299-719300ac-1d6a-11e8-92bb-7f5caa80e263.PNG)

The `<li>` elements you see there? I absolutely cannot, no matter how many hundreds of combinations I have tried, I cannot get them into the json file as two separate text items. It just will not happen. They will either go in as one concatenated item, with no space between, which I guess is OK, and I can regex them apart afterwards, but theoretically I shouldn't have to, right? Shouldn't I be able to grab them by their `<li>` tags somehow, since they are children elements of the preceding levels that are being successfully pulled from the website? 

Two of the many examples:
`li: $('.yt-lockup-meta-info').find('li').text()` returns _all_ of the `<li>` for every meta on the entire website. Since the designer didn't bother to provide any css styling whatsoever at that level, it just returns everything on the page. So I tried:

`li: $('ul .yt-lockup-meta').children('li').text()` and it returns nothing, just `""` . 

FWIW, This is all inside a function that is already successfully returning the `<h3>` parent element; I would think that somehow helps keep it in the family?

```
$('.yt-lockup-content').each(function (i, elem) {
                devtoList[i] = {
                    title: $(this).find('h3').text(),
                    li: $('ul .yt-lockup-meta').children('li').text(),
 }
            });
```

Like I said, I won't bore you with all the hundreds of other combinations I tried, but nothing works. Any suggestions would be very welcome! 

Thanks

**Comment by AaronRobinson** [comment]
I’d use yt-lockup-meta-info > li.nth-of-type(1) to get ‘21 hours ago’

**Comment by I-keep-trying** [comment]
Thanks, yeah, I thought I had resolved this finally today, but still my results are inconsistent. That's not cheerio's fault, that seems to be par for the course with this type of exercise. The same script works and then doesn't work, without changing a single thing sometimes! 

I wouldn't think you would need to do `nth-of-type` since there is only one level of `<li>`, or am I confused? I'll give that a try anyway. 

Here's what ended up working today: 

```
let content = 'div#title-wrapper, ytd-video-meta-block'
     $(content).each(function (i, elem) {
            arr[i] =
            $(elem).find('a#video-title, span').text()
        if (i > 40) {
    return false
}
    })
```

I know that `<span>`  isn't on that screen capture, but for some reason it changed, who knows with youtube. It would have been exactly the same but with `<li>` instead. This grabs all the text, the title and the views plus time posted, which is what I was after, and then I just do string replace to make it readable. 

I'm still not at all certain what the underlying principle is. I started with a really long list of nodes, starting at the very top parent node, going all the way down to the specific element tag, and once I had successfully returned some data, I started deleting them one at a time until I got down to this. It still seems rather arbitrary. The `content` tags are sort of in the middle of the tree, and I was able to skip a level here and there.
